### PR TITLE
Fix EL9 and aarch64 builds

### PR DESCRIPTION
**Pull request by jmontleon**
`-flto=thin` is the updated value for `_clang_lto_cflags` in the `redhat-rpm-config` version `201`, which has the effect of dramatically improving build time. RHEL 9.2 is still on version `199`, so if/until this lands we can override it to get this benefit.

There is also an issue with the older version of ld where it seems to open too many files and linking fails. Increasing the limit a bit works around this.

Interestingly, using the version of ld in go-toolset-13 / clang-16 alone doesn't resolve the issue, but in combination with `-flto=thin` it becomes no longer necessary to increase the maximum number of open files.

Long story short, if/when clang 16 and redhat-rpm-config 201 land in RHEL 9 these can be removed.

### CODE REVIEW

**Comment by jmontleon** [comment]
Fixes https://github.com/trixirt/pytorch-fedora/issues/3

**Comment by jmontleon** [comment]
Successful EPEL9 builds demonstrated here: https://copr.fedorainfracloud.org/coprs/jmontleon/pytorch/build/6239794/
